### ISSUE TITLE
Errors during playing up to freezing

### ISSUE THREAD

**Issue by simonsan** [CorsixTH/CorsixTH#973]
Playing the game sometimes I get errors with an popup that asks me whether I want to recover files.
Here's the output in the console.

```
---------------------------------------------------------------

Welcome to CorsixTH 0.50!

---------------------------------------------------------------

This window will display useful information if an error occurs.

---------------------------------------------------------------

[swscaler @ 000001231e8c9f60] Warning: data is not aligned! This can lead to a speedloss
Checking for CorsixTH updates...
You are running the latest version of CorsixTH.
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:209>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:187: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:137>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:651: in method 'queueAction'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\spawn.lua:39: in field '?'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:642: in method 'setNextAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:435: in field 'setHospital'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:674: in method 'setHospital'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:387: in method 'die'
        C:\Program Files\CorsixTH\Lua\room.lua:795: in local 'remove_humanoid'
        C:\Program Files\CorsixTH\Lua\room.lua:801: in method 'crashRoom'
        C:\Program Files\CorsixTH\Lua\entities\machine.lua:93: in method 'machineUsed'
        ...Files\CorsixTH\Lua\humanoid_actions\multi_use_object.lua:259: in local 'timer_function'
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

Warning: No event handler for timer
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\entities\staff.lua:59: attempt to index a nil value (field 'hospital')
stack traceback:
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:59: in method 'tickDay'
        C:\Program Files\CorsixTH\Lua\world.lua:1161: in method 'onEndDay'
        C:\Program Files\CorsixTH\Lua\world.lua:1076: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The buttonup handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\room.lua:504: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\room.lua:504: in method 'tryAdvanceQueue'
        C:\Program Files\CorsixTH\Lua\entities\machine.lua:212: in method 'setRepairing'
        C:\Program Files\CorsixTH\Lua\dialogs\machine_dialog.lua:125: in field 'callback_ok'
        C:\Program Files\CorsixTH\Lua\dialogs\confirm_dialog.lua:81: in method 'close'
        C:\Program Files\CorsixTH\Lua\dialogs\confirm_dialog.lua:71: in local 'callback'
        C:\Program Files\CorsixTH\Lua\window.lua:664: in method 'handleClick'
        C:\Program Files\CorsixTH\Lua\window.lua:1445: in method 'onMouseUp'
        C:\Program Files\CorsixTH\Lua\window.lua:1429: in field 'onMouseUp'
        C:\Program Files\CorsixTH\Lua\ui.lua:623: in function <C:\Program Files\CorsixTH\Lua\ui.lua:610>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

Warning: No event handler for buttonup
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:209>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in field 'tick'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:149: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>
```

```
---------------------------------------------------------------

Welcome to CorsixTH 0.50!

---------------------------------------------------------------

This window will display useful information if an error occurs.

---------------------------------------------------------------

[swscaler @ 0000022298c9a620] Warning: data is not aligned! This can lead to a speedloss
Checking for CorsixTH updates...
You are running the latest version of CorsixTH.
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:209>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:187: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:137>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:651: in method 'queueAction'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\spawn.lua:39: in field '?'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:642: in method 'setNextAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:435: in field 'setHospital'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:674: in method 'setHospital'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:387: in method 'die'
        C:\Program Files\CorsixTH\Lua\room.lua:795: in local 'remove_humanoid'
        C:\Program Files\CorsixTH\Lua\room.lua:801: in method 'crashRoom'
        C:\Program Files\CorsixTH\Lua\entities\machine.lua:93: in method 'machineUsed'
        ...Files\CorsixTH\Lua\humanoid_actions\multi_use_object.lua:259: in local 'timer_function'
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

Warning: No event handler for timer
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\entities\staff.lua:59: attempt to index a nil value (field 'hospital')
stack traceback:
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:59: in method 'tickDay'
        C:\Program Files\CorsixTH\Lua\world.lua:1161: in method 'onEndDay'
        C:\Program Files\CorsixTH\Lua\world.lua:1076: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:209>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:187: in local 'timer_function'
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in field 'tick'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:149: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\world.lua:1844: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\world.lua:1844: in method 'findRoomNear'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:63: in upvalue 'action_seek_room_find_room'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:213: in field '?'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:667: in method 'finishAction'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:152: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:137>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\world.lua:1844: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\world.lua:1844: in method 'findRoomNear'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:63: in upvalue 'action_seek_room_find_room'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:213: in field '?'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:667: in method 'finishAction'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:152: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:137>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The buttonup handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:60: attempt to call a nil value (local 'timer_function')
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:60: in local 'on_interrupt'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:638: in method 'setNextAction'
        C:\Program Files\CorsixTH\Lua\entities\staff.lua:421: in method 'onClick'
        C:\Program Files\CorsixTH\Lua\ui.lua:627: in function <C:\Program Files\CorsixTH\Lua\ui.lua:610>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>

Warning: No event handler for buttonup
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\world.lua:1844: attempt to index a nil value (field 'queue')
stack traceback:
        C:\Program Files\CorsixTH\Lua\world.lua:1844: in method 'findRoomNear'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:63: in upvalue 'action_seek_room_find_room'
        ...rogram Files\CorsixTH\Lua\humanoid_actions\seek_room.lua:213: in field '?'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:575: in upvalue 'Humanoid_startAction'
        C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:667: in method 'finishAction'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:152: in function <C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:137>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\entity.lua:218: in method 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1121: in method 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:878: in function <C:\Program Files\CorsixTH\Lua\app.lua:875>
        (...tail calls...)
        C:\Program Files\CorsixTH\Lua\app.lua:779: in function <C:\Program Files\CorsixTH\Lua\app.lua:774>
```

**Comment by TheCycoONE** [comment]
Errors like this usually indicate that a path is blocked so someone can't get where they want to go. e.g. there is an open issue about radiators being able to block doors. Can you see if there is anything like that?

**Comment by simonsan** [comment]
You mean I've got to play with administrator privileges? Yeah, there are some kind of "double doors", and people cannot enter or leave a room cause they get stuck at the door trapping everyone inside and creating a bunch of waiting guys outside. Cannot edit the room either, "zZz" won't disappear cause nobody can leave. Picking up the doctor inside creates only another copy of that doctor, that cannot move.

**Comment by MarkL1961** [comment]
No that is not what he said!
Look for radiators in front of doors.  Use transparent walls feature to see other side of doors and walls

**Comment by TheCycoONE** [comment]
@simonsan can you attach a save that reproduces this issue (e.g. upload one to dropbox or another file service and link)

**Comment by TheCycoONE** [comment]
Not enough info, possibly fixed by #1163
